fe/fe-core/src/main/java/com/starrocks/planner/DeltaLakeScanNode.java

```diff
@@ -133,42 +136,45 @@ public List<TScanRangeLocations> getScanRangeLocations(long maxScanRangeLength)
     }
 
     public void setupScanRangeLocations(DescriptorTable descTbl) throws AnalysisException {
-        DeltaLog deltaLog = deltaLakeTable.getDeltaLog();
-        if (!deltaLog.tableExists()) {
-            return;
-        }
-        // use current snapshot now
-        Snapshot snapshot = deltaLog.snapshot();
-        preProcessConjuncts(snapshot.getMetadata().getSchema());
-        List<String> partitionColumnNames = snapshot.getMetadata().getPartitionColumns();
+        Metadata deltaMetadata = deltaLakeTable.getDeltaMetadata();
+
+        preProcessConjuncts(deltaMetadata.getSchema());
+        List<String> partitionColumnNames = deltaLakeTable.getPartitionColumnNames();
         // PartitionKey -> partition id
         Map<PartitionKey, Long> partitionKeys = Maps.newHashMap();
 
-        DeltaScan scan = deltaLakePredicates.isPresent() ? snapshot.scan(deltaLakePredicates.get()) : snapshot.scan();
-
-        for (CloseableIterator<AddFile> it = scan.getFiles(); it.hasNext(); ) {
-            AddFile file = it.next();
-            Map<String, String> partitionValueMap = file.getPartitionValues();
-            List<String> partitionValues = partitionColumnNames.stream().map(partitionValueMap::get).collect(
-                    Collectors.toList());
-
-            Metadata metadata = snapshot.getMetadata();
-            PartitionKey partitionKey =
-                    PartitionUtil.createPartitionKey(partitionValues, deltaLakeTable.getPartitionColumns(),
-                            deltaLakeTable.getType());
-            addPartitionLocations(partitionKeys, partitionKey, descTbl, file, metadata);
+        TableClient tableClient = deltaLakeTable.getTableClient();
+        ScanBuilder scanBuilder = deltaLakeTable.getDeltaSnapshot().getScanBuilder(tableClient);
+        Scan scan = deltaLakePredicates.isPresent() ?
+                scanBuilder.withFilter(tableClient, deltaLakePredicates.get()).build() :
+                scanBuilder.build();
+
+        for (CloseableIterator<FilteredColumnarBatch> scanFilesAsBatches = scan.getScanFiles(tableClient);
```

Comment:
Can we  adapt to deltalake table partition prune and table statistics using this api?

Reply:
yes, getScanFiles will do  partition prune and use stats to skip data